src/mrpro/operators/LinearOperator.py

```
@@ -47,6 +48,59 @@ def H(self) -> LinearOperator:  # noqa: N802
+        max_iterations: int = 64,
+    ) -> torch.Tensor:
+        """Power iteration for computing the operator norm of the linear operator.
+
+        Parameters
+        ----------
+        initial_value
+            initial value to start the iteration; if chosen exactly as zero-tensor,
+            we randomly generate an initial value.
+        dim
+            the dimensions of the tensors on which the operator operates.
+            For example, for a matrix-vector multiplication example, a batched matrix tensor with shape (4,30,80,160),
+            input tensors of shape (4,30,160) to be multiplied, and dim = None, it is understood that the the
+            matrix representation of the operator corresponds to a block diagonal operator (with 4*30 matrices)
+            and thus the algorithm returns a tensor of shape (1,1,1) containing one single value.
+            In contrast, if for example, dim=(-1,), the algorithm computes a batched operator
+            norm and returns a tensor of shape (4,30,1) corresponding to the operator norms of the respective
+            matrices in the diagonal of the block-diagonal operator (if considered in matrix representation).
+            In any case, the output of the algorithm has the same number of dimensions as the elements of the
+            domain of the considered operator (whose dimensionality is implicitly defined by choosing dim), such that
+            the pointwise multiplication of the operator norm and elements of the domain (to be for example used
+            in a Landweber iteration) is well-defined.
+        max_iterations, optional
+            maximal number of iterations
+
+        Returns
+        -------
+            an estimaton of the operator norm
+        """
+        # check if initial value is exactly zero; if yes, set it to a random value
+        vector = torch.randn_like(initial_value) if (initial_value == 0.0).all() else initial_value
+
+        for _ in range(max_iterations):
```

should we raise an exception if max_iterations is smaller then 1?